src/main/java/uk/gov/gchq/palisade/service/request/DataRequestResponse.java

```diff
@@ -44,79 +43,81 @@ public DataRequestResponse() {
     }
 
     public DataRequestResponse token(final String token) {
-        requireNonNull(token, "The token cannot be null.");
-        this.token = token;
+        this.setToken(token);
         return this;
     }
 
-    public void setToken(final String token) {
-        token(token);
+    public DataRequestResponse resource(final LeafResource resource, final ConnectionDetail connectionDetail) {
+        this.setResource(resource, connectionDetail);
+        return this;
     }
 
-    public String getToken() {
-        requireNonNull(token, "The token has not been set.");
-        return token;
+    public DataRequestResponse resources(final Map<LeafResource, ConnectionDetail> resources) {
```

Comment:
this mutator method needs @generated annotation adding

Reply:
Done